### ISSUE TITLE
[b3.sh] setting LOG_ENABLED="1" does not create log

### ISSUE THREAD

**Issue by thomasleveil** [BigBrotherBot/big-brother-bot#284]
when setting `LOG_ENABLED="1"` no log file is created, but if the log file exists then it gets written to.

It seems this is due to [`readlink -f`](https://github.com/BigBrotherBot/big-brother-bot/blob/release-1.10/scripts/b3.sh#L72) which set `LOG_FILE` to the empty string.

I would be tempted to remove the `readlink -f` line all together but as it is not clear to me what purpose is has here I prefer opening this issue to discuss the proper way to deal with this.

**Comment by danielepantaleone** [comment]
The problem was due to git not committing empty directories :smile:
Basically the log file is just the console output being logged to a file that I placed in a subdirectory: b3/scripts/log/b3_init.log. But the directory *log* was missing and thus readlink was returning an empty string and so log was not being written.
I added a check inside the log function to check whether the folder is there, else create it.
I like to keep readlinks, also if the script changes to the *scripts* directory upon execution so relative paths would work. But since it's a script people may want to edit it, and if they remove the line which does the directory change (`cd $(cd -P -- "$(dirname -- "$0")" && pwd -P)`), everything would be messed up.

**Comment by thomasleveil** [comment]
Instead of using `readlink -f` and changing directory, it seems safer to set a `SCRIPT_DIR` variable and use that variable later on:

```shell

SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
# ...
# LOG_FILE="$(readlink -f "${LOG_PATH}")"
LOG_FILE="${SCRIPTDIR}/${LOG_PATH}"
```

this way you don't either have to worry about eventual symbolic links (hence the no `-P`)

**Comment by danielepantaleone** [comment]
I'll provide this change later

Inviato da iPhone

**Comment by danielepantaleone** [comment]
Implemented in #288 
I also ported your changes (https://github.com/BigBrotherBot/big-brother-bot/commit/b301b77a84b35a1f8b392909a3fb696bf61d6a40 and https://github.com/BigBrotherBot/big-brother-bot/commit/646b030ccf0ae29960607ea46237a73f68acbee0)
